project/webapp/view/ProductsList.view.xml

```diff
@@ -1,7 +1,209 @@
-<mvc:View controllerName="veronchi.leverx.project.controller.ProductsList"
-    xmlns:mvc="sap.ui.core.mvc" displayBlock="true"
-    xmlns="sap.m">
-    <Page id="products-list" title="{i18n>ProductsListTitle}">
-        <content />
-    </Page>
-</mvc:View>
+<mvc:View
+  controllerName="veronchi.leverx.project.controller.ProductsList"
+  xmlns:mvc="sap.ui.core.mvc"
+  displayBlock="true"
+  xmlns="sap.m"
+  xmlns:f="sap.f"
+  xmlns:core="sap.ui.core"
+  xmlns:fb="sap.ui.comp.filterbar"
+>
+  <f:DynamicPage
+    id="productsListPage"
+    headerPinned="true"
+    toggleHeaderOnTitleClick="true"
+  >
+    <f:title>
+      <f:DynamicPageTitle>
+        <f:heading>
+          <Title text="{i18n>ProductsPageTitle}" />
+        </f:heading>
+      </f:DynamicPageTitle>
+    </f:title>
+    <f:header>
+      <f:DynamicPageHeader>
+        <f:content>
+          <fb:FilterBar
+            id="filterbar"
+            useToolbar="false"
+            showFilterConfiguration="false"
+            showGoOnFB="false"
+            showClearOnFB="true"
+            clear="onClearFilters"
+          >
+            <fb:filterGroupItems>
+              <fb:FilterGroupItem
+                name="Name"
+                label="Product Name"
+                groupName="Group1"
+                visibleInFilterBar="true"
+              >
+                <fb:control>
+                  <SearchField
+                    id="searchName"
+                    placeholder="Enter product name"
+                    showSearchButton="false"
+                    liveChange="onSearchProducts"
+                  />
+                </fb:control>
+              </fb:FilterGroupItem>
+
+              <fb:FilterGroupItem
+                name="Category"
+                label="Category"
+                groupName="Group1"
+                visibleInFilterBar="true"
+              >
+                <fb:control>
+                  <MultiComboBox
+                    id="categorySelect"
+                    name="Category"
+                    items="{
+                                        path: 'filterBarModel>/categories',
+                                        templateShareable: true
```

Comment:
it's better to use false here
read article about templateShareble https://sapui5.hana.ondemand.com/#/topic/3a4a9e562988456c9be0ef883ae7da50